packages/sparql-proxy/test/index.test.js

```diff
@@ -0,0 +1,173 @@
+import { expect } from 'chai'
+import trifidCore, { getListenerURL } from 'trifid-core'
+import rdf from '@zazuko/env-node'
+import sinon from 'sinon'
+import sparqlProxy from '../index.js'
+
+describe('sparql-proxy', () => {
+  let trifidListener
+
+  async function startTrifid(config) {
+    const server = await trifidCore({
+      server: {
+        listener: {
+          port: 4242,
+        },
+        logLevel: 'warn',
+      },
+    }, {
+      sparqlProxy: {
+        module: sparqlProxy,
+        config: {
+          endpointUrl: 'http://example.com/sparql',
+          serviceDescriptionWorkerUrl: new URL('./support/workerDouble.js', import.meta.url),
```

Comment:
I struggled a little with mocking the fetch inside worker thread. Especially because of two execution contexts but also `rdf.fetch` uses `node-fetch` which is incompatible with native streams.

Instead, I decided to add a configuration option with the worker path and in tests I load worker modules which mock the function to fetch the data.